src/legacy/core_plugins/console/np_ready/public/interfaces/token_iterator.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Token } from './token';
+import { Position } from './core_editor';
+
+export interface TokenIterator {
+  /**
+   * Report the token under the iterator's cursor.
+   */
+  getCurrentToken(): Token | null;
```

Comment:
Should we use class getters for those?

Reply:
(in response to your suggestion about getters above too)

This interface is designed to be an in-place replacement for https://github.com/ajaxorg/ace/blob/master/lib/ace/token_iterator.js.

This reduces the surface area for changes for now. But otherwise I agree, `getters` would be a nice addition/change!

Reply:
Got it, thanks for explaining